app/models/piece.rb

```diff
@@ -33,9 +33,21 @@ def is_obstructed?(x, y, x_target, y_target)
         end
       end
     elsif vertical_move?(x, y, x_target, y_target)
-      
+      (y...y_target).each do |y|
+        if game.square_occupied?(y)
+          return true
+        else
+          return false
+        end
+      end 
     elsif diagonal_move?(x, y, x_target, y_target)
-       
+       (coordinate(x, y)...coordinate(x_target, y_target)).each do |square|
```

Comment:
I'm not sure if/how ranges work with arrays. We should set up a test though to see if it does!